api/src/routes/challenge.ts

```diff
@@ -379,5 +379,48 @@ export const challengeRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/save-challenge',
+    {
+      schema: schemas.saveChallenge,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          void reply.code(403);
+          return formatValidationError(
+            error.validation,
+            'That challenge type is not savable.'
+          );
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const saveChallenge = await fastify.prisma.user.update({
+          where: { id: req.session.user.id },
+          select: {
+            savedChallenges: true
+          },
```

Comment:
this isn't query the challenges, this is query the user if they have savedchallenge, use utili.